docs/reference/run.md

```diff
@@ -1385,30 +1385,71 @@ it will provide a named alias for the container being linked to.
 
 ### ENV (environment variables)
 
-When a new container is created, Docker will set the following environment
+On Linux, when a new container is created, Docker will set the following environment
```

Comment:
Looking if we can;

a) combine the "On linux" and "On Windows" parts, so that the "On Windows" is not overlooked
b) better describe that this is talking about the _daemon_ (i.e., on Docker for Windows, these vars are still set if "native windows containers" is not selected